src/datachain/lib/dc.py

```diff
@@ -1509,12 +1511,31 @@ def from_records(
         session = Session.get(session)
         catalog = session.catalog
 
+        if not to_insert and not schema:
+            raise ValueError("Schema is required for creating empty dataset")
```

Comment:
message seems to be wrong or misleading (?) considering the condition `Schema is required for creating empty dataset`. Condition should be `if not schema` then

Reply:
Changed message, please check if now it seems better